src/lib/FlightTasks/tasks/FlightTask.hpp

```diff
@@ -89,7 +101,14 @@ class FlightTask : public control::Block
 	 * To be called regularly in the control loop cycle to execute the task
 	 * @return true on success, false on error
 	 */
-	virtual bool update() = 0;
+	virtual bool update()
+	{
+		updateOutput(_ctrl_setpoints);
+
+		return true;
+	}
+
+	virtual void updateOutput(ControlSetpoint &setpoint) = 0;
```

Comment:
We actually had a message containing the setpoint in private and a setter before and practise with all non-trivial tasks showed that it's not flexible enough hence every task will store it's own copy of setpoints before it all gets set with the setter in the end which just works around the concept creating overhead.

Reply:
@MaEtUgR No, the values can be written directly into the struct, no need for copying.
The main difference is that it's clear to any developer where the outputs are written in contrast to the old implementation where the outputs could be overwritten anywhere.

Reply:
It's the same like @Stifael writes below: https://github.com/PX4/Firmware/pull/9064#discussion_r174165479
You reverted back to the original design which we changed just very recently with https://github.com/PX4/Firmware/pull/8933/commits/da97ae67054cb4cba6ca056bec16e6b4b582c6d8 and adapted all tasks in https://github.com/PX4/Firmware/pull/8933/commits/c089cb79399810c09518059b5a9a18368195452d because as the commit description says it was not enough flexible. Every task had it's local setpoints which were overwritten at any point and the highest common task then wrote all the outputs which is just more complicated and has more overhead than when the base task directly holds the setpoints and they are not copied one more time.